### PR TITLE
feat: Upgrade goavro version and AvroProducer implementation

### PR DESCRIPTION
**Pull request by cobolbaby**
- Upgrade goavro version
- Upgrade AvroProducer implementation
- Fix Error: `io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Subject not found.; error code: 40401`

### CODE REVIEW

**Comment by cobolbaby** [comment]
@dangkaka Could you accept the merge request?

File: avroProducer.go [hunk 1]
```diff
@@ -74,3 +73,29 @@ func (ap *AvroProducer) Add(topic string, schema string, key []byte, value []byt
...
 	ac.producer.Close()
 }
+
+// AvroEncoder encodes schemaId and Avro message.
+type AvroEncoder struct {
+	SchemaID int
+	Content  []byte
+}
+
+// 此处无数坑，因为Confluent schema registry正对Avro序列化规则有特殊要求，不光需要序列化具体的内容，还要附加上Schema ID以及Magic Byte
```
Comment (dangkaka) [inline avroProducer.go]: please replace with english comment

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module github.com/cobolbaby/go-kafka-avro
```
Comment (dangkaka) [inline go.mod]: module name is wrong

**Review by dangkaka — CHANGES_REQUESTED** [review]
hi @cobolbaby

I left some small notes

File: avroProducer.go (hunk 1 above)
Comment (cobolbaby) [reply avroProducer.go]: Sorry, I have changed.

File: go.mod (hunk 2 above)
Comment (cobolbaby) [reply go.mod]: Sorry, I have changed.

**Review by dangkaka — APPROVED** [review]
LGTM 👍 
thanks @cobolbaby

File: avroProducer.go [hunk 3]
```diff
@@ -27,7 +33,7 @@ func NewAvroProducer(kafkaServers []string, schemaRegistryServers []string) (*Av
 
 //GetSchemaId get schema id from schema-registry service
 func (ap *AvroProducer) GetSchemaId(topic string, avroCodec *goavro.Codec) (int, error) {
-	schemaId, err := ap.schemaRegistryClient.CreateSubject(topic, avroCodec)
+	schemaId, err := ap.schemaRegistryClient.CreateSubject(topic+"-value", avroCodec)
```
Comment (dangkaka) [inline avroProducer.go:36]: oops sorry @cobolbaby 
but why do we need `-value`?

Comment (cobolbaby) [reply avroProducer.go:36]: Fix Error: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Subject not found.; error code: 40401

Comment (cobolbaby) [reply avroProducer.go:36]: Ref: https://stackoverflow.com/questions/53731566/kafka-connect-jdbcsinkconnector-getting-error-retrieving-avro-schema-for-id

Comment (dangkaka) [reply avroProducer.go:36]: ah thanks

Comment (dangkaka) [reply avroProducer.go:36]: @cobolbaby can you please help to fix the wrong unittest?
```
--- FAIL: TestAvroProducer_Add (0.00s)
    /Users/user/works/dangkaka/go-kafka-avro/avroProducer_test.go:18: Error adding msg: unexpected end of JSON input
    /Users/user/works/dangkaka/go-kafka-avro/sync_producer.go:115: Expected to exhaust all expectations, but 1 are left.
FAIL
FAIL	github.com/dangkaka/go-kafka-avro	0.013s
Error: Tests failed.
```
a mismatch between the topic name with `-test`

**Pull request merged**